### ISSUE TITLE
ScrollArea component violates CSP directive rules

### ISSUE THREAD

**Issue by darlantc** [mantinedev/mantine#3597]
### What package has an issue

@mantine/core

### Describe the bug

We use strict CSP rules in our project and we configured Mantine to work with it using emotion cache:
```typescript
const emotionCache = createEmotionCache({ nonce: cspNonce });
<MantineProvider emotionCache={emotionCache}>
...
</MantineProvider>
```
Everything is working as expected, but where we render the Select / Autocomplete component and then we click to open the Dropdown the following error occurs:
![CleanShot 2023-02-22 at 17 58 13](https://user-images.githubusercontent.com/5101009/220757575-a4895d1d-3f61-49d0-8271-699bd9625a31.png)

I can't see any style differences between my localhost and the production builds, so I'm not sure which styles are being blocked by it.

### What version of @mantine/hooks page do you have in package.json?

5.10.4

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by cyantree** [comment]
To get to the bottom of this I've disabled closing of the `Select` component and then compared the complete DOM from the closed with the opened state.

I found the following style tag that only exists in the opened state:
```html
<style>[data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}</style>
```

So in this case the dependency `@radix-ui/react-scroll-area` might not support these CSP rules.

However I also found the following style tags that were already there in the closed state but don't have a nonce set:
```html
<style data-emotion="css-global" data-s=""></style>
<style data-emotion="css-global" data-s=""></style>
<style data-emotion="mantine" data-s=""></style>
```

I didn't checked this in an actual CSP configured environment yet but will do so shortly.

**Comment by rtivital** [comment]
@cyantree do you know what attributes we need to add to the style tag to fix this issue?

**Comment by cyantree** [comment]
Yes. We just need to add `nonce={nonce}` to the style attribute where `{nonce}` is the one configured in the emotion cache.

```
const emotionCache = createEmotionCache({key: 'app', nonce: '1234567890'});

<style nonce="1234567890">
```

I've found another style tag that misses the nonce (when configured):
https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/use-scroll-lock/utils/make-style-tag.ts

This line is also problematic as it requires an `unsafe-inline` configuration to work:
https://github.com/mantinedev/mantine/blob/master/src/mantine-rte/src/modules/image-uploader/ImageUploader.ts#L32

Besides those two findings (from `@radix-ui/react-scroll-area` and `use-scroll-lock`) the real world application I used for testing worked with `style-src 'self' 'nonce-1234567890'`.

**Comment by rtivital** [comment]
Great, thanks for research! So basically, we need to provide a way to set `nonce` attribute on `<style />` tag?

**Comment by cyantree** [comment]
What do you mean exactly?

- For `@radix-ui/react-scroll-area` we/someone must reach out to the devs and open an issue there. I think the solution might be for them to add a prop for configuring the nonce. Or they find another approach that doesn't require a style tag at all.
- For `useScrollLock()` it must retrieve the configured nonce from `emotionCache` (maybe via `useEmotionCache()`)  and add it via `.setAttribute()` to the tag. Maybe via `makeStyleTag({nonce})` or afterwards via `sheet.setAttribute()`.
- For `ImageUploader` the style might be set through `this.fileHolder.style` instead of setting it as attribute. But I haven't looked at this by now.

**Comment by darlantc** [comment]
Any news about it? Is there something I can do to fix the problem?

**Comment by cyantree** [comment]
@darlantc If you want to you can reach out to `@radix-ui/react-scroll-area` as part of the problem lies in this upstream issue and I don't know any way how we might work around it.

The other points can directly be fixed in Mantine so you can jump right into it if you want. (If it's ok for @rtivital)

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.

**Comment by rtivital** [comment]
ScrollArea was migrated to native Mantine implementation, Mantine no longer depends on Radix UI for ScrollArea component
